### DISCUSSION TITLE
Method Filament\Forms\Components\CheckboxList::pivotData does not exist. Need Help

### DISCUSSION THREAD

**Discussion by syed-reza98** [filamentphp/filament#14199] [Help]
### Package

Form builder

### Package Version

v3.2.0

### How can we help you?

BadMethodCallException
PHP 8.1.10
10.48.20
Method Filament\Forms\Components\CheckboxList::pivotData does not exist.
 https://filamentphp.com/docs/3.x/forms/fields/checkbox-list#saving-pivot-data-to-the-relationship

![image](https://github.com/user-attachments/assets/3d2b9064-7e39-4259-b201-c5641b0c8b7e)

In My App\Models\TopSheet.php file I have:
```
    public function topSheetUser(): BelongsToMany
    {
        return $this->belongsToMany(User::class, 'top_sheet_user')->withPivot([
            'total_travel',
        ])->withTimestamps();
    }
```
    
In My App\Filament\Resources\TopSheetResource.php I have:
```
    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\CheckboxList::make('Top Sheet Users')
                    ->relationship(name: 'topSheetUser', titleAttribute: 'name')
                    ->searchable()
                    ->getOptionLabelFromRecordUsing(fn (Model $record) => "{$record->name} ({$record->email})")
                    ->pivotData([
                        'total_travel' => 0.00,
                    ])
                    ->bulkToggleable(),
            ]);
    }
```
    
My composer.json is:

```
    "require": {
        "php": "^8.1",
        "filament/filament": "3.2",
        "filament/forms": "3.2",
        "guzzlehttp/guzzle": "^7.2",
        "laravel/framework": "^10.10",
        "laravel/sanctum": "^3.2",
        "laravel/tinker": "^2.8",
        "spatie/laravel-permission": "^6.1"
    },

```    
 Kindly help me with this problem.

**Comment by e2d2-dev** [comment]
looks like you try to do something easy way too difficult.what are you trying to achive?

**Reply by syed-reza98** [reply]
For now, I want to attach multiple users to a top_sheet using the top_sheet_user pivot table along with additional column data.

Using a relational manager is an easy approach. However, the people for whom I’m developing this project may not understand the attach/detach actions.

Instead, a checkbox list would be much more understandable for them.

If you could please provide a solution to this issue, it would be quite helpful.

**Comment by e2d2-dev** [comment]
had a similar issue, solved it by using a repeater (forms/infolists) and used hydrating/dehydrating.
don't using many to many relationships and doing pivot models also helps. (create a pivot model, it's in the docs). then you're just using a one to many relationship and stuff gets way easier.
thats a design issue and needs further introduction with the changellange itself. more info needed

**Reply by syed-reza98** [reply]
I haven’t created a pivot model yet. I’ll follow your suggestion and let you know if it works. Thank you.
